app/src/main/java/com/lagradost/quicknovel/providers/ArchiveOfOurOwnProvider.kt

```diff
@@ -0,0 +1,187 @@
+package com.lagradost.quicknovel.providers
+
+import android.annotation.SuppressLint
+import com.lagradost.quicknovel.*
+import com.lagradost.quicknovel.MainActivity.Companion.app
+import com.lagradost.quicknovel.mvvm.debugWarning
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import java.lang.Exception
+import java.util.*
+import kotlin.collections.ArrayList
+
+class ArchiveOfOurOwnProvider : MainAPI() {
+    override val name = "Archive of Our Own"
+    override val mainUrl = "https://archiveofourown.org"
+
+    override val hasMainPage = true
+
+    override val iconId = R.drawable.ic_archive_of_our_own
+
+    override val iconBackgroundId = R.color.royalRoadColor
+
+    override val orderBys = listOf(
+        Pair("Latest", "latest"),
+    )
+    val searchFilters = listOf(
+        Pair("Best Match", "_score"),
+        Pair("Author", "authors_to_sort_on"),
+        Pair("Title", "title_to_sort_on"),
+        Pair("Date Posted", "created_at"),
+        Pair("Date Updated", "revised_at"),
+        Pair("Word Count", "word_count"),
+        Pair("Hits", "hits"),
+        Pair("Kudos", "kudos_count"),
+        Pair("Comments", "comments_count"),
+        Pair("Bookmarks", "bookmarks_count")
+
+    )
+
+
+    override val tags = listOf(
+        "Creator Chose Not To Use Archive Warnings",
+        "Graphic Depictions Of Violence",
+        "Major Character Death",
+        "Rape/Non-con",
+        "Underage",
+        "Creator Chose Not To Use Archive Warnings"
+    ).map { Pair(it, java.net.URLEncoder.encode(it.replace("/","*s*"), "utf-8")) }
+
+
+    override suspend fun loadMainPage(
+        page: Int,
+        mainCategory: String?,
+        orderBy: String?,
+        tag: String?,
```

Comment:
tag and orderBy not used, while being declared above?

Reply:
I thought tag and orderBy were used for search; AO3's main page doesn't filter fics by tag.